compiler/rustc_data_structures/src/profiling.rs

```diff
@@ -649,11 +649,34 @@ impl Drop for VerboseTimingGuard<'_> {
     fn drop(&mut self) {
         if let Some((start_time, start_rss, ref message)) = self.start_and_message {
             let end_rss = get_resident_set_size();
-            print_time_passes_entry(&message, start_time.elapsed(), start_rss, end_rss);
+            let dur = start_time.elapsed();
+
+            if should_print_passes(dur, start_rss, end_rss) {
+                print_time_passes_entry(&message, dur, start_rss, end_rss);
+            }
         }
     }
 }
 
+fn should_print_passes(
+    dur: Duration, 
+    start_rss: Option<usize>,
+    end_rss: Option<usize>,
+) -> bool {
+    if dur.as_millis() > 5 {
+        return true;
+    }
+    
+    if let (Some(start_rss), Some(end_rss)) = (start_rss, end_rss) {
+        let change_rss = end_rss as i128 - start_rss as i128;
+        if change_rss.is_positive() {
+            return true;
+        }
```

Comment:
This works but I think it would be clearer to use [`usize::abs_diff()`](https://doc.rust-lang.org/std/primitive.usize.html#method.abs_diff) instead.

Reply:
Will do.